tests/components/history/test_websocket_api.py

```diff
@@ -1584,3 +1584,233 @@ async def test_overflow_queue(
+            "start_time": now.isoformat(),
+            "entity_ids": ["sens!or.one", "two"],
+            "include_start_time_state": True,
+            "significant_changes_only": False,
+            "no_attributes": True,
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"] is False
+    assert "invalid_entity_ids" in str(response)
+
+    await client.send_json(
+        {
+            "id": 5,
+            "type": "history/stream",
+            "start_time": now.isoformat(),
+            "entity_ids": ["sensor.one", "sensortwo."],
+            "include_start_time_state": True,
+            "significant_changes_only": False,
+            "no_attributes": True,
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"] is False
+    assert "invalid_entity_ids" in str(response)
+
+    await client.send_json(
+        {
+            "id": 6,
+            "type": "history/stream",
+            "start_time": now.isoformat(),
+            "entity_ids": ["one", ".sensortwo"],
+            "include_start_time_state": True,
+            "significant_changes_only": False,
+            "no_attributes": True,
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"] is False
+    assert "invalid_entity_ids" in str(response)
+
+
+async def test_history_during_period_for_invalid_entity_ids(
+    recorder_mock: Recorder, hass: HomeAssistant, hass_ws_client: WebSocketGenerator
+) -> None:
+    """Test history_during_period for valid and invalid entity ids."""
+    now = dt_util.utcnow()
+
+    await async_setup_component(hass, "history", {})
+    await async_setup_component(hass, "sensor", {})
+    await async_recorder_block_till_done(hass)
+    hass.states.async_set("sensor.one", "on", attributes={"any": "attr"})
+    await async_recorder_block_till_done(hass)
+    hass.states.async_set("sensor.two", "off", attributes={"any": "attr"})
+    await async_wait_recording_done(hass)
+
+    await async_wait_recording_done(hass)
+
+    client = await hass_ws_client()
+
+    await client.send_json(
+        {
+            "id": 1,
+            "type": "history/history_during_period",
+            "start_time": now.isoformat(),
+            "include_start_time_state": True,
+            "significant_changes_only": False,
+            "no_attributes": True,
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"]
+    assert "sensor.one" in str(response)
+    assert "sensor.two" in str(response)
+
+    await client.send_json(
+        {
+            "id": 2,
+            "type": "history/history_during_period",
+            "start_time": now.isoformat(),
+            "entity_ids": ["sensor.one"],
+            "include_start_time_state": True,
+            "significant_changes_only": False,
+            "no_attributes": True,
+        }
+    )
+    response = await client.receive_json()
+    assert response["success"]
```

Comment:
Since the response is structured data we should assert its in the right place in the response instead of stringifying the response

Reply:
Fair enough :)